src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationServiceRunner.vb

```diff
@@ -404,9 +404,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
                 End If
             End If
 
+            If _context.IsAsyncMemberDeclarationContext Then
+                Return symbols.WhereAsArray(Function(s) IsSymbolValidForAsyncDeclarationContext(s))
+            End If
+
             Return symbols
         End Function
 
+        Private Shared Function IsSymbolValidForAsyncDeclarationContext(symbol As ISymbol) As Boolean
+            Dim namedType As INamedTypeSymbol = TryCast(symbol, INamedTypeSymbol)
+            Return namedType IsNot Nothing AndAlso
+                namedType.TypeKind = TypeKind.Class AndAlso
+                namedType.Name = "Task" AndAlso
+                namedType.TypeParameters.Length < 2 AndAlso
+                namedType.IsFromSystemRuntimeOrMscorlibAssembly()
```

Comment:
none of this seems VB sepcific.  also, see prior messages about this not following IDE general patterns for matching against types.

Reply:
Sholud I create some `IsAwaitableTask` extension method to reuse it in both C# and VB or what?

Reply:
that seems reasonable.